### PR TITLE
fix(sns-subscriptions): restrict encryption of queue to only the respective sns topic (under feature flag)

### PR DESCRIPTION
**Pull request by daschaa**
Implements #20339 

The change adds a conditional that only the corresponding SNS Topic can decrypt the SQS queue with the KMS-Key if one was given.

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/master/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by daschaa** [comment]
@kaizencc Could you review this PR please? I would really appreciate your feedback :)

**Review by corymhall — CHANGES_REQUESTED** [review]
@daschaa, unfortunately since this does technically introduce a breaking change, this will need to live behind a [feature flag](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md#feature-flags).

**Comment by daschaa** [comment]
@corymhall I have now added the feature flag. Since this is my first time implementing a feature flag, it would be cool if you could double check everything and give some feedback if it is ok like it is implemented. 
For the unit tests, i could not get the `testFutureBehavior` and `testLegacyBehavior` to work. My solution was to just pass it in myself as context when instantiating a new App. I hope it is ok?
Thanks in advance!

File: packages/@aws-cdk/cx-api/lib/features.ts [hunk 1]
```diff
@@ -282,6 +282,17 @@ export const CODEPIPELINE_CROSS_ACCOUNT_KEY_ALIAS_STACK_SAFE_RESOURCE_NAME = '@a
  */
 export const S3_CREATE_DEFAULT_LOGGING_POLICY = '@aws-cdk/aws-s3:createDefaultLoggingPolicy';
 
+/**
+* Enable this feature flag to restrict the decryption of a SQS queue, which is subscribed to a SNS topic, to
```
Comment (corymhall) [inline packages/@aws-cdk/cx-api/lib/features.ts:286]: Can you also add this info [here](https://github.com/aws/aws-cdk/blob/68f09b7c6f8e6c1ddf13fdd4e116d12333d24c46/packages/%40aws-cdk/cx-api/README.md?plain=1#L14)? We haven't found the best way to document these yet so until then I figure that will have to do.

**Review by corymhall — DISMISSED** [review]
@daschaa thanks for implementing the feature flag! I just have a couple of cosmetic comments and then we are good to go!

File: packages/@aws-cdk/aws-sns-subscriptions/lib/sqs.ts [hunk 2]
```diff
@@ -51,11 +52,20 @@ export class SqsSubscription implements sns.ITopicSubscription {
     // if the queue is encrypted, add a statement to the key resource policy
     // which allows this topic to decrypt KMS keys
     if (this.queue.encryptionMasterKey) {
-      this.queue.encryptionMasterKey.addToResourcePolicy(new iam.PolicyStatement({
+      let policy: iam.PolicyStatementProps = {
```
[the remark this answers is not in the document]
Comment (daschaa) [reply packages/@aws-cdk/aws-sns-subscriptions/lib/sqs.ts:55]: Ah yes! Way better! I totally forgot that the ternary operator could be used for map values :) Thanks for the hint!

File: packages/@aws-cdk/cx-api/lib/features.ts (hunk 1 above)
Comment (daschaa) [reply packages/@aws-cdk/cx-api/lib/features.ts:286]: If it is ok I will add a chore later and update the feature-flag documentation :)

**Comment by daschaa** [comment]
@corymhall Thanks for the review! I have added everything (I think) ❤️

**Review by corymhall — APPROVED** [review]

**Pull request merged**